dirty_cat/super_vectorizer.py

```diff
@@ -169,45 +168,61 @@ def __init__(self, *,
         self.verbose = verbose
 
     @staticmethod
-    def _auto_cast(X: pd.DataFrame) -> pd.DataFrame:
+    def _auto_cast(X) -> pd.DataFrame:
         """
-        Takes a pandas DataFrame and tries to convert its columns to the best
+        Takes an array-like or a dataframe and tries to convert its columns to the best
         possible data type.
 
         Parameters
         ----------
-        X: pd.DataFrame
-            Input data as a pandas DataFrame.
+        X : {array-like, dataframe} of shape (n_samples, n_features)
+            The data to be transformed.
 
         Returns
         -------
         array
             The same array, with its columns casted to the best possible
             data type.
-            Columns with missing values won't be modified.
         """
         from pandas.core.dtypes.base import ExtensionDtype
+        # To use TextParser, we need to have a list of rows
+        colnames = None  # save the column names which we lose during the conversion
+        if isinstance(X, pd.DataFrame):
+            colnames = X.columns
+            X = X.to_numpy().tolist()
+        elif isinstance(X, np.ndarray):
+            X = X.tolist()
+
+        parser = pd.io.parsers.TextParser(X,
+                                          header=None,
+                                          na_values=[None, " ", "?", "..."])  # additional values to be considered as NA
+        X = parser.read()
+
+        for col in X.columns:
+            # only convert string, object or categorical columns to datetimes
+            if X[col].dtype != "int64" and X[col].dtype != "float64":
```

Comment:
Should it be useful to account for columns with type `int32` or similar ?
Somewhere in the code, I've used [is_numeric_dtype](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.is_numeric_dtype.html), perhaps it could also make the code easier to read ?